### PR TITLE
Deploy scripts

### PR DESCRIPTION
**Pull request by gunzip**
- Added script to deploy functions to the Azure cloud
- Added docs (README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,116 @@
...
+
+There a few options to deploy the Functions to Azure.
+
+### Using GitHub
+
+If you have a GitHub account you can set up a [continuos deployment flow](https://docs.microsoft.com/en-us/azure/azure-functions/functions-continuous-deployment).
+
+1. Fork the GitHub repository into your account.
+
+2. Edit scripts/local.config.js and write up your GitHub settings:
```
Comment (cloudify) [inline README.md]: use backticks for file paths

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "express": "^4.15.3",
     "nodemailer": "^4.0.1",
     "nodemailer-sendgrid-transport": "^0.2.0",
+    "shelljs": "^0.7.8",
```
Comment (cloudify) [inline package.json]: move this to the `devDependencies` block

File: scripts/resources.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
+const $ = require('shelljs')
```
Comment (cloudify) [inline scripts/resources.js]: no shebang? is this script meant to be executed?

Comment (cloudify) [inline scripts/resources.js]: also the script needs a comment block explaining it's purpose and usage

File: scripts/resources.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+const program = require('commander')
+const CONF = require('./local.config')()
+
+// die on first error
+$.config.fatal = true
+
+// VERBOSE=1 node deploy.js
+$.config.verbose = !!process.env.VERBOSE || false
```
Comment (cloudify) [inline scripts/resources.js]: there are two problems with this statement :)

`!!"0" === true`

also the `|| false` is unnecessary (`A || false === A`)

File: scripts/resources.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+// Create a resource group if not exists
+// az group create \
+//       --name $resourceGroupName \
+//       --location $location
+
```
Comment (cloudify) [inline scripts/resources.js]: a sanity check would be nice:

```
if (!shell.which('az')) {
  shell.echo('Sorry, this script requires the Azure management cli to be installed');
  shell.exit(1);
}
```

File: scripts/resources.js (hunk 3 above)
Comment (cloudify) [inline scripts/resources.js]: also `$` is not a very nice name for a variable, perhaps `shell` ?

File: scripts/resources.js [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    --resource-group ${CONF.resourceGroupName} \
+    --slot-settings "SENDGRID_KEY=${CONF.sendGridKey}"`)
+}
+
+//////////////////// FUNCTIONS (git deployment)
+
+// az functionapp deployment user set --user-name $username --password $password
+
+const deploy_functionDeployementSource = () => {
+  if (CONF.gitRepoUrl !== '') {
```
Comment (cloudify) [inline scripts/resources.js]: also `CONF.gitRepoUrl != null`

File: scripts/resources.js (hunk 3 above)
Comment (gunzip) [inline scripts/resources.js]: about the shebang, it does not work in win32 environments (without cygwin / mingw), so
i suggest to run the script using node from command line (which is cross platform)

File: scripts/resources.js (hunk 5 above)
Comment (gunzip) [inline scripts/resources.js]: done.

**Comment by gunzip** [comment]
see https://github.com/teamdigitale/digital-citizenship/pull/14

**Pull request closed without merging**